tests/topotests/ospf_prefix_suppression/test_ospf_prefix_suppression.py

```diff
@@ -0,0 +1,959 @@
+    test_func = partial(
+        topotest.router_json_cmp, r3, "show ip route 10.1.8.0/24 json", input_dict, True
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=1)
+    assertmsg = "10.1.8.0/24 installed on router r3"
+    assert result is None, assertmsg
+
+
+def test_broadcast_stub_suppression():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip("Skipped because of router(s) failure")
+
+    step("Configure R1 interface r1-eth4 with prefix suppression")
+    r1 = tgen.gears["r1"]
+    r1.vtysh_multicmd("conf t\ninterface r1-eth4\nip ospf prefix-suppression")
+
+    step("Verify the R1 configuration of 'ip ospf prefix-suppression'")
+    prefix_suppression_cfg = (
+        tgen.net["r1"]
+        .cmd('vtysh -c "show running ospfd" | grep "^ ip ospf prefix-suppression"')
+        .rstrip()
+    )
+    assertmsg = "'ip ospf prefix-suppression' applied, but not present in configuration"
+    assert prefix_suppression_cfg == " ip ospf prefix-suppression", assertmsg
+
+    step("Verify that ospf-prefix suppression is applied to the R1 interface")
+    r1_eth4_with_prefix_suppression = {
+        "interfaces": {
+            "r1-eth4": {
+                "ifUp": True,
+                "ospfEnabled": True,
+                "ipAddress": "10.1.7.1",
+                "ospfIfType": "Broadcast",
+                "prefixSuppression": True,
+            }
+        }
+    }
+    test_func = partial(
+        topotest.router_json_cmp,
+        r1,
+        "show ip ospf interface r1-eth4 json",
+        r1_eth4_with_prefix_suppression,
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=1)
+    assertmsg = "R1 OSPF interface r1-eth4 doesn't have prefix-suppression enabled"
+    assert result is None, assertmsg
+
+    step(
+        "Verify that ospf-prefix suppression is applied to the R1 interface (non-JSON)"
+    )
+    prefix_suppression_show = (
+        tgen.net["r1"]
+        .cmd(
+            'vtysh -c "show ip ospf interface r1-eth4" | grep "^  Suppress advertisement of interface IP prefix"'
+        )
+        .rstrip()
+    )
+    assertmsg = (
+        "'ip ospf prefix-suppression' applied, but not present in interface show"
+    )
+    assert (
+        prefix_suppression_show == "  Suppress advertisement of interface IP prefix"
+    ), assertmsg
+
+    step("Verify the ospf prefix is not advertised and not present on r3")
+    r3 = tgen.gears["r3"]
+    input_dict = {}
+    test_func = partial(
+        topotest.router_json_cmp, r3, "show ip route 10.1.7.0/24 json", input_dict, True
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=1)
+    assertmsg = "10.1.7.0/24 installed on router r3"
+    assert result is None, assertmsg
+
+    step("Remove R1 interface r1-eth4 prefix-suppression configuration")
+    r1 = tgen.gears["r1"]
+    r1.vtysh_multicmd("conf t\ninterface r1-eth4\nno ip ospf prefix-suppression")
```

Comment:
nit: vtysh_cmd() is doing the same, but a generic wrapper.

Reply:
I changed all these, ran the test, and it works fine.  So I will use the vtysh_cmd().